daemon/graphdriver/devmapper/deviceset.go

```diff
@@ -1657,6 +1686,9 @@ func (devices *DeviceSet) Shutdown() error {
 	var devs []*devInfo
 
 	devices.Lock()
+	// Save DeviceSet Metadata first
```

Comment:
docker kills all the threads if shutdown does not finish with-in certain time (I think 10 seconds). So if some device deactivation takes a long time (as device is busy), then it is possible that docker kills the Shutdown() routine and we never get a chance to save deviceset metadata. That's why save it first before moving on to deactivating devices. 

I will put this information in comments.